### PR TITLE
feat(@desktop/chat): download messages

### LINKED ISSUE #3508: option to export a chat log to a json format, so we can compare chat logs
- this should be a toggle in the advanced settings that should make available an option to export channel logs in the channel menu
- the exported log should be a json file that includes for each message
  - an id/hash
  - the message
  - the user who sent it
  - timestamp
  - lamport timestamp

### PR DESCRIPTION
**Pull request by alaibe**
fixes #3508

https://github.com/status-im/nimqml/pull/34
https://github.com/status-im/dotherside/pull/52

### CODE REVIEW

File: src/app/chat/views/messages.nim [hunk 1]
```diff
@@ -480,3 +480,16 @@ QtObject:
...
+    for message in self.messageList[self.channelView.activeChannel.id].messages:
+      if message.id == "":
+        continue
+
+      messages.elems.add(%*{
+        "id": message.id, "text": message.text, "clock": message.clock,
+        "alias": message.alias, "from": message.fromAuthor
+      })
+
+    writeFile(replace(filePath, "file://", ""), $messages)
```
Comment (pinebit) [inline src/app/chat/views/messages.nim:495]: Please don't do any magic with `file://`, please please use QUrl::toLocalFile and/or QUrl::fromLocalFile if you need a conversion between QUrl and native disk filepath.

Comment (pinebit) [reply src/app/chat/views/messages.nim:495]: There is another pending PR here https://github.com/status-im/status-desktop/pull/3534#issuecomment-919432339 having the same issue. Maybe it is time to expose these two conversion functions from DOtherSide?

Comment (alaibe) [reply src/app/chat/views/messages.nim]: @pinebit updated pls let me know if that's better for you!

**Review by pinebit — APPROVED** [review]

File: src/app/chat/views/messages.nim (hunk 1 above)
Comment (pinebit) [reply src/app/chat/views/messages.nim]: Perfect! Thanks!

**Review by iurimatias — APPROVED** [review]

**Pull request merged**